src/transformers/generation/utils.py

```diff
@@ -1510,7 +1510,7 @@ def _tensor_or_none(token_kwargs, token_self, device=None):
             logger.warning(f"Setting `pad_token_id` to `eos_token_id`:{pad_token_id} for open-end generation.")
 
         # we can't infer attn mask if pad token is set to be eos token in model's generation config
-        if eos_token_id is not None and torch.isin(elements=eos_token_id, test_elements=pad_token_id).any():
```

Comment:
cc @gante Are there any reasons e.g. compilation for using torch.isin?

Reply:
yes, compilation requires this `torch.isin` 💔 cc @sanchit-gandhi 

imo, we should create an internal function containing [this torch.isin](https://github.com/huggingface/transformers/blob/main/src/transformers/generation/stopping_criteria.py#L490) workaround (that works on compile AND mps devices), and replace all `torch.isin` call by this function